### ISSUE TITLE
[Grid - Basics] Column headers are announced incorrectly for Column Groups

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3190]
## Current behavior

In [Column Groups](https://storybook.sanomalearning.design/?path=/story/grid-basics--column-groups) when screen reader user go from header to header they are announced that "Name" and "School" spans for only one column. That makes more complication in table, because then header "Name" is connected with "First name" and "School" is connected with "Last name". Other three headers has no connection with headers above them. 
This is confusing to users and conveys false information.

https://github.com/user-attachments/assets/0a9424e4-04be-41f3-b927-0f366868420f

https://github.com/user-attachments/assets/c547b486-fdf5-4e60-abc2-9b4008359629

## Expected behavior

Screen reader should announce what is the span of first two headers.
VoiceOver should announce connection between headers for first header that is in the scope (so: "Name and First Name" but after that "Last Name" is read alone because it's still in "Name" scope).
NVDA should announce connection between headers for every header.

https://github.com/user-attachments/assets/3a3dad48-4cac-44e9-a96d-43f97a00a29c

https://github.com/user-attachments/assets/1d30d7b6-31d5-4632-aa53-ecd9bd857842

This problem seems to be result of something different than incorrect HTML because HTML looks as in best practices examples and it also works when opened in codepen.

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
